### PR TITLE
feat: set metadata during getBackups and getMetadata calls

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/session.ts [hunk 1]
```diff
@@ -264,28 +264,49 @@ describe('Session', () => {
...
       };
       session.getMetadata(gaxOptions, assert.ifError);
     });
+
+    it('should update metadata', done => {
+      const metadata = {};
+      session.request = (config, callback) => {
+        callback(null, metadata);
+      };
+      session.getMetadata(() => {
```
Comment (stephenplusplus) [inline test/session.ts:294]:
```suggestion
      session.getMetadata((err) => {
        assert.ifError(err);
```

File: src/instance.ts [hunk 2]
```diff
@@ -355,7 +355,9 @@ class Instance extends common.GrpcServiceObject {
         let backupInstances: Backup[] | null = null;
         if (backups) {
           backupInstances = backups.map(backup => {
-            return this.backup(backup.name!);
+            const backupInstance = this.backup(backup.name!);
+            backupInstance.metadata = backup;
```
Comment (stephenplusplus) [inline src/instance.ts:359]: Nice catch. Please add a test to see that this is set.

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

File: src/instance.ts (hunk 2 above)
Comment (AVaksman) [reply src/instance.ts:359]: I believe I added that verification in an existing [test](https://github.com/googleapis/nodejs-spanner/pull/1198/files#diff-189d5720e2e29b973f957a72855ced1cR1459). Is it sufficient?

File: test/session.ts (hunk 1 above)
Comment (AVaksman) [reply test/session.ts]: Done, thanks!

File: src/instance.ts (hunk 2 above)
Comment (stephenplusplus) [reply src/instance.ts:359]: 🤦 sorry. LG!

**Review by stephenplusplus — APPROVED** [review]

**Pull request merged**